### PR TITLE
implement support for access controls directly on binary resources

### PR DESCRIPTION
**Pull request by acoburn**
See: https://jira.duraspace.org/browse/FCREPO-1791

### CODE REVIEW

File: src/test/java/org/fcrepo/auth/webac/WebACRolesProviderTest.java [hunk 1]
```diff
@@ -126,13 +126,13 @@ public void noAclTest() throws RepositoryException {
 
         when(mockResource.getPath()).thenReturn(accessTo);
         when(mockResource.getContainer()).thenReturn(mockParentResource);
-        when(mockResource.getTriples(anyObject(),
-                    eq(PropertiesRdfContext.class))).thenReturn(new RdfStream());
+        when(mockResource.getTriples(anyObject(), eq(PropertiesRdfContext.class)))
```
Comment (awoods) [inline src/test/java/org/fcrepo/auth/webac/WebACRolesProviderTest.java]: Is there anything actually changed here? I'm squinting... anything?

File: src/test/java/org/fcrepo/auth/webac/WebACRolesProviderTest.java [hunk 2]
```diff
@@ -126,13 +126,13 @@ public void noAclTest() throws RepositoryException {
...
+                .thenReturn(new RdfStream());
         when(mockNode.getDepth()).thenReturn(1);
 
         when(mockParentResource.getNode()).thenReturn(mockParentNode);
-        when(mockParentResource.getTriples(anyObject(),
-                    eq(PropertiesRdfContext.class))).thenReturn(new RdfStream());
+        when(mockParentResource.getTriples(anyObject(), eq(PropertiesRdfContext.class)))
+                .thenReturn(new RdfStream());
```
Comment (awoods) [inline src/test/java/org/fcrepo/auth/webac/WebACRolesProviderTest.java]: same here??

File: src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java [hunk 3]
```diff
@@ -299,9 +307,10 @@ public void deleteRoles(final Node node) throws RepositoryException {
                     child.getTriples(translator, PropertiesRdfContext.class)
                          .filter(p -> isAclPredicate.test(model.asStatement(p).getPredicate()))
                          .forEachRemaining(t -> {
-                            aclTriples.putIfAbsent(t.getPredicate().getURI(), new ArrayList<>());
+                             aclTriples.putIfAbsent(t.getPredicate().getURI(), new ArrayList<>());
```
Comment (awoods) [inline src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java]: Please rollback format update.

File: src/test/java/org/fcrepo/auth/webac/WebACRolesProviderTest.java (hunk 1 above)
Comment (acoburn) [inline src/test/java/org/fcrepo/auth/webac/WebACRolesProviderTest.java]: nope, just reformatted to make the line consistent with everything else in the file. shall I revert that?

File: src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java [hunk 4]
```diff
@@ -299,9 +307,10 @@ public void deleteRoles(final Node node) throws RepositoryException {
...
                              if (t.getObject().isURI()) {
-                                aclTriples.get(t.getPredicate().getURI()).add(t.getObject().getURI());
+                                 aclTriples.get(t.getPredicate().getURI()).add(
+                                     substringBeforeLast(t.getObject().getURI(), "/" + JCR_CONTENT));
```
Comment (awoods) [inline src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java]: Obviously it is a little frightening that outside of the kernel we are seeing "jcr:content". It appears to work... but is unsettling nonetheless.

Comment (acoburn) [inline src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java]: Very much agreed.

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4/fcrepo-module-auth-webac/commit/fbd4520d22a9c511af07044401af82384ace11f4

**Pull request closed without merging**